src/Pos/Update/Arbitrary/Core.hs

```diff
@@ -52,7 +52,7 @@ instance Arbitrary UpdateProposal where
                   upSoftwareVersion
                   upData
                   upAttributes
-        pure UpdateProposal {..}
+        pure UnsafeUpdateProposal {..}
```

Comment:
I think `UnsafeUpdateProposal` usage should be avoided as much as possible. I find it better to write `either (error . mapped "mkUpdateProposal failed: ") pure $ mkUpdateProposal d r a t u t' i`, because if something invalid is passed to `mkUpdateProposal`, the error will appear earlier and will be more evident. Also you don't need to duplicate signing logic in this case.